lib/ansible/modules/cloud/vmware/vmware_vnic.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': ''}
+
+DOCUMENTATION = '''
+---
+module: vmware_vnic
+short_description: Manage Operations on vmware vnics
+'''
+
+EXAMPLES = '''
+- name: Remove Vnic
+  vmware_vnic:
+    hostname=10.x.x.x
+    username=xxxx
+    password=xxxx
+    port=443
+    name=vmname
+    uuid=vmuuid
+    nic_state=delete
+    nic_number=2
+'''
+
+import atexit
+import requests
+from tools import cli
+from pyVmomi import vim
+from pyVim.connect import SmartConnect, Disconnect
+from tools import tasks
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_text, to_native
+from ansible.module_utils.vmware import (find_obj, gather_vm_facts, get_all_objs, compile_folder_path_for_object, serialize_spec,vmware_argument_spec, set_vm_power_state, PyVmomi)
+
+# disable  urllib3 warnings
+if hasattr(requests.packages.urllib3, 'disable_warnings'):
+    requests.packages.urllib3.disable_warnings()
+
+
+def update_virtual_nic_state(si, vm_obj, nic_number, new_nic_state):
+
+    nic_prefix_label = 'Network adapter '
+    nic_label = nic_prefix_label + str(nic_number)
+    virtual_nic_device = None
+    for dev in vm_obj.config.hardware.device:
+        if isinstance(dev, vim.vm.device.VirtualEthernetCard) \
+                and dev.deviceInfo.label == nic_label:
+            virtual_nic_device = dev
+    if not virtual_nic_device:
+        raise RuntimeError('Virtual {} could not be found.'.format(nic_label))
+
+    virtual_nic_spec = vim.vm.device.VirtualDeviceSpec()
+    virtual_nic_spec.operation = \
+        vim.vm.device.VirtualDeviceSpec.Operation.remove \
+        if new_nic_state == 'delete' \
+        else vim.vm.device.VirtualDeviceSpec.Operation.edit
```

Comment:
Can we add new NICs as well ?

Reply:
@Akasurde : Thanks for the valuable suggestions. I will update as suggested.

Currently we can delete , connect and disconnect vnics but later we can add functionality to add vnics as well.